### ISSUE TITLE
[ERROR] Failed to fix nested if: Expected outer if to have indented body and no else

### ISSUE THREAD

**Issue by LefterisJP** [charliermarsh/ruff#2094]
ruff version: `0.0.230`

```python
a = True
b = True

if a:
    if b:
        print('do something')  # noqa: T201
else:
    print('at else')  # noqa: T201
```

Running ruff here will result with:

```
test.py:4:1: SIM102 Use a single `if` statement instead of nested `if` statements
```

If we try to autofix we get `[2023-01-22][23:09:59][ruff::rules::flake8_simplify::rules::ast_if][ERROR] Failed to fix nested if: Expected outer if to have indented body and no else`

**Comment by charliermarsh** [comment]
Apart from the autofix error, this is a false positive, right?!

**Comment by charliermarsh** [comment]
Looks like a regression introduced in #2050 \cc @harupy

**Comment by charliermarsh** [comment]
I'll either fix or revert that for now

**Comment by LefterisJP** [comment]
> Apart from the autofix error, this is a false positive, right?!

It's a really tricky false positive.

I think if you "fix" to this code:

```python
if a and b:
    print('do something')  # noqa: T201
else:
    print('at else')  # noqa: T201
```

Then only if `a=True` and `b=False` you get different program flow. Original code would not print anything, while fixed code would go in the else.

On the other hand, the places where it showed up in our code, did require some refactoring, so I don't count it as a total loss.

**Comment by charliermarsh** [comment]
Yeah, I do consider it a false positive because the suggested fix actually has different behavior and semantics.

**Comment by charliermarsh** [comment]
Ok, this should be fixed in the next release (either tonight or tomorrow morning).
